File: Rotate array.cpp
//Problem Link: https://leetcode.com/problems/rotate-array/


class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k%=nums.size();
        
        for(int i=0;i<nums.size()/2;i++)
            swap(nums[i],nums[nums.size()-1-i]);
        
        for(int i=0;i<k/2;i++)
            swap(nums[i],nums[k-1-i]);
        
        for(int i=0;i<nums.size();i++)
            cout<<nums[i]<<" ";
        
        int i=k,j=nums.size()-1,num=(nums.size()-k)/2;
        while(num--){
            swap(nums[i++],nums[j--]);
        }
         
    }
};

File: Rotate array.js
//https://leetcode.com/problems/rotate-array/


/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    k%=nums.length;
    num=nums.splice(nums.length-k,k);
    for(let i=0;i<nums.length;i++)
        num.push(nums[i]);
    nums.splice(0,nums.length);
    for(let i=0;i<num.length;i++)
        nums.push(num[i]);
};